src/mscorlib/src/System/StringComparer.cs

```diff
@@ -213,83 +213,77 @@ internal sealed class CultureAwareComparer : StringComparer
 
     }
 
-    // Provide x more optimal implementation of ordinal comparison.
+    // Provide a more optimal implementation of ordinal comparison.
     [Serializable]
     internal sealed class OrdinalComparer : StringComparer 
 #if FEATURE_RANDOMIZED_STRING_HASHING           
         , IWellKnownStringEqualityComparer
 #endif
     {
-        private bool            _ignoreCase;
-       
-        internal OrdinalComparer(bool ignoreCase) {
-               _ignoreCase = ignoreCase;
-        }
-
-        public override int Compare(string x, string y) {
-            if (Object.ReferenceEquals(x, y)) return 0;
-            if (x == null) return -1;
-            if (y == null) return 1;
-            
-            if( _ignoreCase) {
-                return String.Compare(x, y, StringComparison.OrdinalIgnoreCase);
-            }
-                    
-            return String.CompareOrdinal(x, y);                                
-        }
-                
-        public override bool Equals(string x, string y) {
-            if (Object.ReferenceEquals(x ,y)) return true;
-            if (x == null || y == null) return false;
+        public override int Compare(string x, string y) =>
+            object.ReferenceEquals(x, y) ? 0 :
+            x == null ? -1 :
+            y == null ? 1 :
+            string.CompareOrdinal(x, y);
```

Comment:
@jkotas, I saw a related comment from you on another PR... what's your opinion, for example, of whether to have:
```C#
public override bool Equals(string x, string y) => string.Equals(x, y);
```
vs:
```C#
public override bool Equals(string x, string y) => ReferenceEquals(x, y) || string.Equals(x, y);
```
vs
```C#
public override bool Equals(string x, string y) => ReferenceEquals(x, y) || (x != null && y != null && string.Equals(x, y));
```
i.e. whether to delegate to string.Equals for everything or special-case reference equality and null to avoid the static call to Equals.